p4-16/spec/P4-16-spec.mdk

```diff
@@ -2425,6 +2426,124 @@ struct Parsed_headers {
 }
 ~ End P4Example
 
+### Structure overlays { #sec-overlays }
+
+An overlay allows the access to fields or subfields of a `struct` or
+`header` using alternative names.  An overlay is constructed by a
+special `overlay` "function", that establishes a connection between
+the fields of the source and destination structure.
+
+The following example shows an use of overlays:
+
+~ Begin P4Example
+// In this header all fields are byte-aligned
+header IPv4_aligned {
+    bit<8>       version_ihl;
+    bit<8>       diffserv;
+    bit<16>      totalLen;
+    bit<16>      identification;
+    bit<16>      flags_fragOffset;
+    bit<8>       ttl;
+    bit<8>       protocol;
+    bit<16>      hdrChecksum;
+    bit<32>      srcAddr;
+    bit<32>      dstAddr;
+}
+
+// This overlay introduces a structure type named IPv4
+overlay IPv4 expand(IPv4_aligned s) {
+    bit<4>  version = s.version_ihl[7:4];
+    bit<4>  ihl = s.version_ihl[3:0];
+    bit<8>  diffserv = s.diffserv;
+    bit<16> totalLen = s.totalLen;
+    bit<16> identification = s.identification;
+    bit<3>  flags = s.flags_fragOffset[15:13];
+    bit<13> fragOffset = s.flags_fragOffset[12:0];
+    bit<8>  ttl = s.ttl;
+    bit<8>  protocol = s.protocol;
+    bit<16> hdrChecksum = s.hdrChecksum;
+    bit<32> srcAddr = s.srcAddr;
+    bit<32> dstAddr = s.dstAddr;
+}
+
+parser P(...) {
+   IPv4_aligned header;
+   IPv4 fields = expand(header);
+   ...
+   state ipv4 {
+      packet.extract(header);
+      verify(fields.version == 5, error.UnsupportedVersion);
+   }
+}
+~ End P4Example
+
+It is expected that such coding style will make it easier to generate
+efficient implementations on some architectures.
+
+The `_ = _;` notation indicates that all fields of a source which have
+not yet been referenced are mapped unchanged to corresponding fields
+of the destination.  Using this notation, the previous overlay can be
+simplified as:
+
+~ Begin P4Example
+overlay IPv4 expand(IPv4_aligned s) {
+    bit<4>  version = s.version_ihl[7:4];
+    bit<4>  ihl = s.version_ihl[3:0];
+    bit<3>  flags = s.flags_fragOffset[15:13];
+    bit<13> fragOffset = s.flags_fragOffset[12:0];
+    _ = _; // map all remaining fields unchanged
+}
+~ End P4Example
+
+An overlay must define all fields of the destination structure.  The
```

Comment:
I am not sure, but was the sentence "An overlay must define all fields of the destination structure." applicable when this PR was originally written, when the overlay required a separate struct definition for the overlay, separate from the struct or header that was overlaid?

Given that this has been removed, this sentence seems like it might be obsolete.  If it is not obsolete, what does the sentence mean?